cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedWorkflowProgramRunner.java

```diff
@@ -207,6 +238,16 @@ private Resources max(Resources r1, Resources r2) {
                          Math.max(vcores1, vcores2));
   }
 
+  private Resources sumMemory(Resources r1, Resources r2) {
+    // Fix to use one vcore
```

Comment:
Is this because we know the workflow driver won't use much cpu? It seems weird that we can potentially use fewer vcores for a workflow with mr1 and mr2 in parallel versus a workflow with just mr1.

It seems like we should either always use 1 vcore for the driver, or use the max vcores of r1 and r2 here.

Reply:
I was thinking about this one what is the realistic number. It definitely not summed as in memory, as usually cores are under utilize in the workflow driver, unless the job is actually doing some heavy computation task on the client side. I am setting it to `1` such that if the user does need more vcores, it can be set directly via the `task.workflow.system.resources.vcores` setting.

Probably a better estimate would be something like 1 core for every 4 branches in a fork. What do you think?

Reply:
If we use the greater of r1.vcores and r2.vcores, I think that guarantees that we'll never use fewer vcores in a branched workflow than we would if we re-arranged everything to be serial.

I suppose we could argue that vcores should actually be summed if the branches contain actions/conditions, since they all run in the workflow driver.

This is probably over-analyzing though, as I think 1 vcore is enough for the vast majority of situations, and in other situations the user should probably explicitly set the number. I don't have a strong opinion, but would favor doing the max for forks.

Reply:
Updated to use max of vcores